### PR TITLE
isProviderSupported check

### PR DESCRIPTION
**Pull request by mrajashree**
During upgrade, add a check to see if the initially configured auth provider
is supported by auth-service before creating the generic_object

### CODE REVIEW

**Comment by mrajashree** [comment]
https://github.com/rancher/rancher/issues/9448

File: server/auth_server.go [hunk 1]
```diff
@@ -614,19 +614,26 @@ func UpgradeCase() error {
...
+		log.Errorf("UpgradeCase: Error reading DB settings %v", err)
 		return err
 	}
 
 	// Get all provider specific settings from setting table for previously configured auth control
 	for _, val := range providers.Providers {
+		if !providers.IsProviderSupported(val) {
+			log.Debug("Auth provider not supported by rancher-auth-service")
+			return nil
+		}
```
Comment (prachidamle) [inline server/auth_server.go]: providers.Providers is a list in rancher-auth-service itself, right? I dont see the reason why we need this check while iterating that list

Comment (mrajashree) [reply server/auth_server.go]: @prachidamle oh yes, this check isn't needed here

File: server/auth_server.go [hunk 2]
```diff
@@ -624,9 +624,12 @@ func UpgradeCase() error {
 		if err != nil {
 			return err
 		}
+		if p == nil {
+			return fmt.Errorf("UpgradeCase: Could not get the %s auth provider", val)
+		}
 		pSettings, err := readCommonSettings(p.GetProviderSettingList(false))
 		if err != nil {
-			log.Errorf("GetConfig: Error reading DB settings for previous auth providers %v", err)
+			log.Errorf("UpgradeCase: Error reading DB settings for previous auth providers %v", err)
```
Comment (cjellick) [inline server/auth_server.go]: It is a bad pattern to log an error and then return one. Use errors.Wrap() to wrap the error and provide a new error message and just return it.

The general pattern across all languages is: only log the error if you are handling and _not_ allowing it to bubble up further.

File: server/auth_server.go [hunk 3]
```diff
@@ -643,23 +646,31 @@ func UpgradeCase() error {
...
 
+		if newProvider == nil {
+			return fmt.Errorf("UpgradeCase: Could not get the %s auth provider", config.Provider)
+		}
+
 		config.AllowedIdentities = getAllowedIdentities(dbSettings[allowedIdentitiesSetting], "", newProvider.GetIdentitySeparator())
 		providerSettings, err := readCommonSettings(newProvider.GetProviderSettingList(false))
 		if err != nil {
-			log.Errorf("GetConfig: Error reading provider DB settings %v", err)
+			log.Errorf("UpgradeCase: Error reading provider DB settings %v", err)
```
Comment (cjellick) [inline server/auth_server.go]: as above, wrap and return, don't log

File: server/auth_server.go [hunk 4]
```diff
@@ -643,23 +646,31 @@ func UpgradeCase() error {
...
 			return err
 		}
 		newProvider.AddProviderConfig(&config, providerSettings)
 
 		provider, err = initProviderWithConfig(&config)
 		if err != nil {
-			log.Errorf("UpdateConfig: Cannot update the config, error initializing the provider %v", err)
+			log.Errorf("UpgradeCase: Cannot update the config, error initializing the provider %v", err)
```
Comment (cjellick) [inline server/auth_server.go]: same as above

**Pull request merged**